packages/synapse-react-client/src/components/ChallengeSubmission/ChallengeSubmission.tsx

```diff
@@ -0,0 +1,281 @@
+    enabled: isLoggedIn && !!challenge,
+    onSettled: (
+      data: PaginatedIds | undefined,
+      error: SynapseClientError | null,
+    ) => {
+      console.log('useGetUserSubmissionTeams', { data }, { error })
+      if (data) {
+        const isReg = data.results.length > 0
+        if (!isReg) {
+          setErrorMessage(
+            'Error: Please join a Submission Team before continuing.',
+          )
+          return
+        }
+        if (data.results.length > 1) {
+          setErrorMessage(
+            'Error: You are a member of more than one Submission Team. You may only belong to one Submission Team per Challenge.',
+          )
+          return
+        }
+        console.log({ setSubmissionTeamId: data.results[0] })
+        setSubmissionTeamId(data.results[0])
+      }
+      if (error) {
+        setErrorMessage(
+          `Error: Could not determine if you are already registered for this Challenge.`,
+        )
+      }
+    },
+  })
+
+  const { data: submissionTeam } = useGetTeam(submissionTeamId!, {
+    enabled: !!submissionTeamId,
+    refetchInterval: Infinity,
+    useErrorBoundary: true,
+  })
+
+  useGetEntityAlias(newProject?.alias ?? EMPTY_ID, {
+    enabled: newProject !== undefined && !!challenge && !!submissionTeam,
+    onSettled: (
+      data: EntityId | null | undefined,
+      error: SynapseClientError | null,
+    ) => {
+      console.log({ newProject })
+      if (data) {
+        setProjectAliasFound(true)
+        setChallengeProjectId(data.id)
+      }
+      if (error) {
+        setProjectAliasFound(false)
+      }
+      console.log('useGetEntityAlias', { data }, { error })
+    },
+  })
+
+  /**
+   * If the challenge project was just created, retrieve its ACL
+   * and add the submission team to it
+   */
+  useGetEntityACL(challengeProjectId ?? EMPTY_ID, {
+    enabled: !!challengeProjectId && isProjectNewlyCreated === true,
+    refetchInterval: Infinity,
+    useErrorBoundary: true,
+    onSettled: (
+      data: AccessControlList | undefined,
+      error: SynapseClientError | null,
+    ) => {
+      console.log('useGetEntityACL', { data }, { error })
+      if (data) {
+        // Give submission team admin access to challenge project
+        const teamResourceAccess: ResourceAccess = {
+          principalId: Number(submissionTeam!.id),
+          accessType: [
+            ACCESS_TYPE.CHANGE_PERMISSIONS,
+            ACCESS_TYPE.CHANGE_SETTINGS,
+            ACCESS_TYPE.CREATE,
+            ACCESS_TYPE.DELETE,
+            ACCESS_TYPE.DOWNLOAD,
+            ACCESS_TYPE.MODERATE,
+            ACCESS_TYPE.READ,
+            ACCESS_TYPE.UPDATE,
+          ],
+        }
+        updateACL({
+          ...data,
+          resourceAccess: [...data.resourceAccess, teamResourceAccess],
+        })
+        setIsProjectNewlyCreated(false)
+      }
+      if (error) {
+        setErrorMessage(`Error: Could not retrieve project ACL.`)
```

Comment:
Is this needed, if you have an error boundary and you've told it to useErrorBoundary?

Reply:
Removed